plugin/src/main/java/org/opennms/timeseries/cortex/CortexTSS.java

```diff
@@ -448,6 +458,44 @@ protected static String tagsToQuery(final Collection<Tag> tags) {
         return b.toString();
     }
 
+    protected static String tagMatchersToQuery(final Collection<TagMatcher> tagMatchers) {
+        StringBuilder b = new StringBuilder();
+        for (TagMatcher matcher : tagMatchers) {
+            String key;
+            String value;
+            if (IntrinsicTagNames.name.equals(matcher.getKey())) {
+                key = METRIC_NAME_LABEL;
+                // TODO: Patrick: sanitation is deactivated since we kill otherwise the regex. We need to find a better solution for that...
+                value = matcher.getValue(); // sanitizeMetricName(tag.getValue());
```

Comment:
Not sure what we should do about that. Ideas?
We don't really need it since we only query for the resourceId via regex => I could change the test.
But it would break the interface contract (ugly)...

Reply:
For the metric name, let's sanitize the value if the match is `=` or `!=`. Let's leave it as-is if it's a regular expression.

We can document this on the API contract: regular expression behavior is undefined for intrinsic tags

Reply:
done.